functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/pages/BasePage.groovy

```diff
@@ -42,19 +49,113 @@ abstract class BasePage {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(60))
+        wait.until {ExpectedConditions.elementToBeClickable(locator)}
+    }
+
+    void waitForTextToBePresentInElement(WebElement locator, String text) {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(60))
+        wait.until {ExpectedConditions.textToBePresentInElement(locator, text)}
+    }
+
+    boolean waitForElementAttributeToChange(WebElement locator, String attribute, String valueCompare) {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(60))
+        wait.until {
+            WebDriver d ->
+                def elementLocator = locator.getAttribute(attribute)
+                elementLocator == valueCompare
+        }
+    }
+
+    void implicitlyWait(int milliSeconds) {
+        context.driver.manage().timeouts().implicitlyWait(Duration.ofMillis(milliSeconds))
+    }
+
+    void waitForNumberOfElementsToBe(By locator, Integer number) {
+        new WebDriverWait(context.driver, Duration.ofSeconds(15))
+                .until(ExpectedConditions.numberOfElementsToBe(locator, number))
+    }
+
+    WebElement waitIgnoringForElementVisible(WebElement locator) {
+        new WebDriverWait(context.driver, Duration.ofSeconds(15))
+                .ignoring(StaleElementReferenceException.class)
+                .until(ExpectedConditions.visibilityOf(locator))
+    }
+
+    WebElement waitIgnoringForElementToBeClickable(WebElement locator) {
+        new WebDriverWait(context.driver, Duration.ofSeconds(15))
+                .ignoring(StaleElementReferenceException.class)
+                .until (ExpectedConditions.elementToBeClickable(locator))
+    }
+
+    WebElement waitForPresenceOfElementLocated(By locator) {
+        new WebDriverWait(context.driver, Duration.ofSeconds(15))
+            .until(ExpectedConditions.presenceOfElementLocated(locator))
+    }
+
+    void waitForUrlToContain(String text) {
+        new WebDriverWait(context.driver, Duration.ofSeconds(15))
+                .until(ExpectedConditions.urlContains(text))
+    }
+
+    boolean waitForAttributeContains(WebElement locator, String attribute, String value) {
+        new WebDriverWait(context.driver, Duration.ofSeconds(15))
+                .until(ExpectedConditions.attributeContains(locator, attribute, value))
+    }
+
+    def waitForModal(int expected) {
+        try {
+            new WebDriverWait(driver, Duration.ofSeconds(15)).until {
+                ExpectedConditions.numberOfElementsToBe(modalField, expected)
+            }
+        } catch (TimeoutException e) {
+            throw new RuntimeException("Timed out waiting for the modal to have ${expected} elements.", e)
         }
     }
 
+    WebElement byAndWait(By locator) {
+        waitForElementVisible locator
+        el locator
+    }
 
     WebDriver getDriver() {
         context.driver
     }
 
+    void executor(String script) {
```

Comment:
a more appropriate name would be "executeScript" (same applies to the method below)